src/dune/file_tree.ml

```diff
@@ -375,7 +390,36 @@ end = struct
         ( Dune_project.ensure_project_file_exists project
           : Dune_project.created_or_already_exist );
       let file = Path.Source.relative path Dune_file.fname in
-      Some (Dune_file.load file ~project)
+      let parents =
```

Comment:
Another way to handle this is to merge `Sub_dirs.Dir_map.` for a directroy and its parent. For example:

```
$ cat dune
(subdir foo
 (subdir bar $1))
$ cat foo/dune
 (rule ..)
```

When we evaluate the dune file for `foo/dune`, we essentially turn it into:

```
(subdir bar $1)
(rule ..)
```

This seems like it should be simpler but will do more work up front.

Reply:
I ended up going this way. It was too hard to reason about the code while having to think about all parent dirs.